### PR TITLE
oauth-token-authn: switch to sha256~ prefix

### PR DESCRIPTION
**Pull request by sttts**
Implementing https://github.com/openshift/enhancements/pull/420.

### CODE REVIEW

File: openshift-kube-apiserver/authentication/oauth/bootstrapauthenticator.go [hunk 1]
```diff
@@ -34,6 +37,14 @@ func NewBootstrapAuthenticator(tokens oauthclient.OAuthAccessTokenInterface, get
 }
 
 func (a *bootstrapAuthenticator) AuthenticateToken(ctx context.Context, name string) (*kauthenticator.Response, bool, error) {
+	// hash token for new-style sha256: prefixed token
```
Comment (stlaz) [inline openshift-kube-apiserver/authentication/oauth/bootstrapauthenticator.go:40]: nit: sha256~

**Comment by sttts** [comment]
Infra and kube unit flakes.

/retest

**Comment by sttts** [comment]
Infra. Sigh.

/retest

**Comment by sttts** [comment]
Infra 😞

**Comment by sttts** [comment]
Errors look all the same. Probably infra.

/retest

**Comment by sttts** [comment]
/override ci/prow/e2e-aws-fips

Unit tests fail due to parallel efforts to get 1.14 going.

**Pull request merged**